lib/api.js

```diff
@@ -109,27 +109,11 @@ var api = {
    * }
    * ```
    */
-  messages: []
-}
+  this.messages = []
 
-/**
- * Extends the tree with the it's API
- *
- * @module API
- *
- * @param   {Array} tree - PostHTML Tree
- *
- * @returns {Array} tree - PostHTML Tree (extended)
- */
-module.exports = function (tree) {
-  tree.walk = api.walk
-  tree.match = api.match
-  tree.messages = api.messages
+  this.extendApi = true
```

Comment:
Can't we just check if  e.g `walk/match` extist ? `this.extendAPI` => `this.isExtended` ?

Reply:
it is not trivial today to `walk/match` there is no tomorrow)

Reply:
or `messages` depending if this is sufficient enough?